flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/GenericInMemoryCatalog.java

```diff
@@ -540,25 +536,18 @@ private boolean isPartitionSpecValid(ObjectPath tablePath, CatalogPartitionSpec
 			return false;
 		}
 
-		CatalogTable table =  (CatalogTable) baseTable;
+		CatalogTable table = (CatalogTable) baseTable;
 		List<String> partitionKeys = table.getPartitionKeys();
 		Map<String, String> spec = partitionSpec.getPartitionSpec();
 
 		// The size of partition spec should not exceed the size of partition keys
-		if (partitionKeys.size() < spec.size()) {
-			return false;
-		} else {
-			int size = spec.size();
+		return partitionKeys.size() == spec.size() && spec.keySet().containsAll(partitionKeys);
```

Comment:
This is different in semantics from the original code.

Reply:
Yes, that's intended change